packages/xchain-cosmos/src/client.ts

```diff
@@ -188,6 +189,17 @@ class Client implements CosmosClient, XChainClient {
     }
   }
 
+  getTransactionData = async (txId: string): Promise<Tx> => {
+    try {
+      const mainAsset = this.getMainAsset()
+      const txResult = await this.sdkClient.txsHashGet(txId)
+
+      return getTxsFromHistory([txResult], mainAsset)[0]
```

Comment:
this seems like possible bottleneck in case if resulted array from `getTxsFromHistory` is empty